#
# Patch to allow krb5 to compile using new Solaris Studio 12.6 compiler.
#
# Note that this patch to acx_pthread.m4 can be removed when MIT updates the
# version of that file to the latest version.  I have filed a bug with MIT.
#
# Patch source: in-house
# MIT bug ticket: 8597 acx_pthread.m4 needs to be updated
#
--- a/src/config/ac-archive/acx_pthread.m4
+++ b/src/config/ac-archive/acx_pthread.m4
@@ -157,7 +157,7 @@ for flag in $acx_pthread_flags; do
         # functions on Solaris that doesn't have a non-functional libc stub.
         # We try pthread_create on general principles.
         AC_TRY_LINK([#include <pthread.h>],
-                    [pthread_t th; pthread_join(th, 0);
+                    [pthread_t th = 0; pthread_join(th, 0);
                      pthread_attr_init(0); pthread_cleanup_push(0, 0);
                      pthread_create(0,0,0,0); pthread_cleanup_pop(0); ],
                     [acx_pthread_ok=yes])
